lib/vmpooler/pool_manager.rb

```diff
@@ -289,161 +289,168 @@ def destroy_vm(vm, pool)
+      $vsphere[pool['name']] ||= Vmpooler::VsphereHelper.new
 
+      $threads[pool['name']] = Thread.new do
         loop do
-          # INVENTORY
-          inventory = {}
-          begin
-            base = $vsphere[pool['name']].find_folder(pool['folder'])
+          _check_pool(pool)
+          sleep(5)
+        end
+      end
+    end
 
-            base.childEntity.each do |vm|
-              if
-                (! $redis.sismember('vmpooler__running__' + pool['name'], vm['name'])) &&
-                (! $redis.sismember('vmpooler__ready__' + pool['name'], vm['name'])) &&
-                (! $redis.sismember('vmpooler__pending__' + pool['name'], vm['name'])) &&
-                (! $redis.sismember('vmpooler__completed__' + pool['name'], vm['name'])) &&
-                (! $redis.sismember('vmpooler__discovered__' + pool['name'], vm['name']))
+    def _check_pool(pool)
+      # INVENTORY
+      inventory = {}
+      begin
+        base = $vsphere[pool['name']].find_folder(pool['folder'])
 
-                $redis.sadd('vmpooler__discovered__' + pool['name'], vm['name'])
+        base.childEntity.each do |vm|
+          if
+            (! $redis.sismember('vmpooler__running__' + pool['name'], vm['name'])) &&
+            (! $redis.sismember('vmpooler__ready__' + pool['name'], vm['name'])) &&
+            (! $redis.sismember('vmpooler__pending__' + pool['name'], vm['name'])) &&
+            (! $redis.sismember('vmpooler__completed__' + pool['name'], vm['name'])) &&
+            (! $redis.sismember('vmpooler__discovered__' + pool['name'], vm['name']))
 
-                $logger.log('s', "[?] [#{pool['name']}] '#{vm['name']}' added to 'discovered' queue")
-              end
+            $redis.sadd('vmpooler__discovered__' + pool['name'], vm['name'])
 
-              inventory[vm['name']] = 1
-            end
-          rescue
+            $logger.log('s', "[?] [#{pool['name']}] '#{vm['name']}' added to 'discovered' queue")
           end
 
-          # RUNNING
-          $redis.smembers('vmpooler__running__' + pool['name']).each do |vm|
-            if inventory[vm]
-              if pool['running_ttl']
-                begin
-                  check_running_vm(vm, pool['name'], pool['running_ttl'])
-                rescue
-                end
-              else
-                begin
-                  check_running_vm(vm, pool['name'], '720')
-                rescue
-                end
-              end
-            end
-          end
+          inventory[vm['name']] = 1
+        end
+      rescue
+      end
 
-          # READY
-          $redis.smembers('vmpooler__ready__' + pool['name']).each do |vm|
-            if inventory[vm]
-              begin
-                check_ready_vm(vm, pool['name'], pool['ready_ttl'] || 0)
-              rescue
-              end
+      # RUNNING
+      $redis.smembers('vmpooler__running__' + pool['name']).each do |vm|
+        if inventory[vm]
+          if pool['running_ttl']
```

Comment:
Can this change between loops? If not, can this passed as an argument to `_check_pool` and set `'720'` as the default value? This `if` can then be collapsed.